### PR TITLE
EVG-19086 Introduce option to match attach.artifacts with exact file names

### PR DESCRIPTION
**Pull request by hadjri**
EVG-19086

### Description
Introducing a new `exact_file_names` flag on the attach.artifacts command, which will cause the command to match exact file names rather than performing regex style matching on the entire work directory.
### Testing
Added some new unit tests to the attach artifacts test suite.
### Documentation
Added a piece to the attach.artifacts documentation for this new field and its functionality.

### CODE REVIEW

File: agent/command/attach_artifacts.go
Comment (malikchaya2) [inline agent/command/attach_artifacts.go:89]: Why do we want to ignore the errors for exactFileNames? I don't know if exactFileNames should automatically be considered optional.

File: agent/command/attach_artifacts_test.go
Comment (malikchaya2) [inline agent/command/attach_artifacts_test.go:116]: I see that this is tested here, but I'd still strongly advocate to test this manually in staging as well, since historically file changes have been prone to bugs that were hard to discover from unit tests.

**Review by malikchaya2 — CHANGES_REQUESTED** [review]

File: agent/command/attach_artifacts.go
Comment (hadjri) [reply agent/command/attach_artifacts.go:89]: I put this to make it such that the command won't fail if the files specified in the YAML don't exist, since `ExactFileNames` is explicit about what files to pick and the current way the command works is more forgiving in this regard since it automatically detects files based on the gitignore style matching. 

That being said I'm fine to remove this and have the expectation that the user is going to be sure the files with these exact names are going to exist or else it will fail.

Comment (malikchaya2) [reply agent/command/attach_artifacts.go]: Since this request came from a specific user, I'd suggest consulting them to see which they prefer. 
If they prefer that if behave like optional does, I'd suggest adding that to the docs.

**Review by malikchaya2 — APPROVED** [review]
LGTM after this is tested on staging. 

I also suggest reaching out to the user before merging to get their preference on if it should be treated as optional.

**Comment by hadjri** [comment]
> LGTM after this is tested on staging.
> 
> I also suggest reaching out to the user before merging to get their preference on if it should be treated as optional.

Confirmed [working](https://spruce-staging.corp.mongodb.com/task/sandbox_git_env_test_validate_commit_message_patch_d631380317a8478b879cc27d6171508dea2b11df_65203eba97b1d37768e33633_23_10_06_17_07_17/logs?execution=0) on staging with the new parameter

**Pull request merged**
